components/nimbus/src/error.rs

```diff
@@ -69,6 +69,8 @@ pub enum NimbusError {
 pub enum BehaviorError {
     #[error("Invalid state: {0}")]
     InvalidState(String),
+    #[error("Invalid argument: {0}")]
+    InvalidDuration(String),
```

Comment:
Since this is only used in stateful files I think this should have the "stateful" feature attribute

```suggestion
    #[cfg(feature = "stateful")]
    #[error("Invalid argument: {0}")]
    InvalidDuration(String),
```

Reply:
It seems that there's already a `cfg(feature = "stateful")` on `BehaviorError`!